src/neuro_flow/storage/api.py

```diff
@@ -363,41 +365,54 @@ def project(
         cluster = cluster or self._cluster_name
         org_name = org_name or self._org_name
         owner = owner or self._client.config.username
+        project_name = project_name or self._client.config.project_name_or_raise
+
         if id:
             return ApiProjectStorage(self._raw_client, id=id)
         else:
             assert yaml_id
             return ApiProjectStorage(
                 self._raw_client,
                 args=dict(
-                    yaml_id=yaml_id, cluster=cluster, org_name=org_name, owner=owner
+                    yaml_id=yaml_id,
+                    cluster=cluster,
+                    org_name=org_name,
+                    owner=owner,
+                    project_name=project_name,
                 ),
             )
 
     async def create_project(
         self,
         yaml_id: str,
+        project_name: str,
         cluster: Optional[str] = None,
         org_name: Optional[str] = None,
     ) -> Project:
         return await self._raw_client.create(
             "flow/projects",
             data={
                 "name": yaml_id,
+                "project_name": project_name,
                 "cluster": cluster or self._cluster_name,
                 "org_name": org_name or self._org_name,
             },
             mapper=_parse_project_payload,
         )
 
     def list_projects(
-        self, name: Optional[str] = None, cluster: Optional[str] = None
+        self,
+        name: Optional[str] = None,
+        cluster: Optional[str] = None,
+        project_name: Optional[str] = None,
```

Comment:
should we also support `org_name` here? otherwise it might not be clear which tenant is meant.